### PR TITLE
[Rewards 3.0] Add auto-contribute and recurring contribution cards

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/40563

Screenshot 2024-08-21 at 8 39 46 AM

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/resources/rewards_strings.grdp [hunk 1]
```diff
@@ -818,4 +818,65 @@
...
+    Sites viewed
+  </message>
+  <message name="REWARDS_AC_SITE_LABEL" desc="">
+    Site
+  </message>
+  <message name="REWARDS_AC_TITLE" desc="">
+    Auto-Contribute
+  </message>
+  <message name="REWARDS_BENEFITS_STORE_SUBTEXT" desc="">
+    Get brave branded products.
```
Comment (emerick) [inline components/resources/rewards_strings.grdp:858]: Capital B in Brave?

File: components/resources/rewards_strings.grdp [hunk 2]
```diff
@@ -818,4 +818,65 @@
...
+    Attention
+  </message>
+  <message name="REWARDS_AC_DISABLED_TEXT" desc="">
+    An automatic way to support websites and content creators, based on how much time you spend on their content.
+  </message>
+  <message name="REWARDS_AC_DISABLED_TITLE" desc="">
+    Auto-Contribute is disabled
+  </message>
+  <message name="REWARDS_AC_EMPTY_LIST_TEXT" desc="">
+    Verified Creators that you visit will appear here as you browse.
```
Comment (emerick) [inline components/resources/rewards_strings.grdp:837]: Nit: I think we only capitalize "creators" when it's "Brave Creators" now.

**Review by emerick — APPROVED** [review]

**Comment by Miyayes** [comment]
@zenparsing Is it possible to have the Auto-Contribute info drawer expanded by default?

File: components/resources/rewards_strings.grdp (hunk 1 above)
Comment (Miyayes) [reply components/resources/rewards_strings.grdp:858]: It should be: "Get Brave-branded products."

File: components/resources/rewards_strings.grdp (hunk 2 above)
Comment (Miyayes) [reply components/resources/rewards_strings.grdp:837]: Ah yes, let's make it lowercase: `Verified creators that you visit will appear here as you browse.`

Comment (Miyayes) [reply components/resources/rewards_strings.grdp:837]: @zenparsing I think the current pre3.0 UI, there are instances of "Verified Creator" (as in when it shows verification status in the UI). We can change all of those specific instances to Verified creator" too.

Comment (zenparsing) [reply components/resources/rewards_strings.grdp]: Done

File: components/resources/rewards_strings.grdp (hunk 1 above)
Comment (zenparsing) [reply components/resources/rewards_strings.grdp]: Done.

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**